app-libs/stf/src/getter.rs

```diff
@@ -214,7 +214,15 @@ impl ExecuteGetter for Getter {
 						serde_json::from_str(orders_string).expect("error serializing to JSON");
 					let orders_encoded: Vec<Vec<u8>> = orders.iter().map(|o| o.encode()).collect();
 
-					let leaf_index_u32: u32 = (*leaf_index).into();
+					let leaf_index: usize = match (*leaf_index).try_into() {
+						Ok(index) => index,
+						Err(_) => {
+							info!("Error converting Leaf Index to usize: {})", leaf_index);
+							return None
+						},
+					};
+
+					let leaf_index_u32 = leaf_index as u32;
 					if leaf_index_u32 >= orders.len() as u32 {
```

Comment:
Here, you can actually just use the original leaf index if you rename the variable above and then you don't need to cast `orders.len()`

```rust
if leaf_index >= orders.len()
```

Reply:
done